contracts/libraries/AddressPermissionControl.sol

```diff
@@ -0,0 +1,147 @@
+            delete _stagedPermissionMasks[target];
+            _stagedAddresses.remove(target);
+        }
+    }
+
+    function _revokePermissionInstant(address from, uint8 permissionId) private {
+        uint256 diff = _permissionIdToMask(permissionId);
+        uint256 currentMask = _permissionMasks[from];
+        require(currentMask & diff != 0, ExceptionsLibrary.INVALID_TARGET);
+        _permissionMasks[from] = currentMask & (~diff);
+        if (_permissionMasks[from] == 0) {
+            delete _permissionMasks[from];
+            _addresses.remove(from);
+        }
+    }
+
+    function _revokePermissionsInstant(address from, uint8[] calldata permissionIds) internal {
+        for (uint256 i; i != permissionIds.length; ++i) {
+            _revokePermissionInstant(from, permissionIds[i]);
+        }
+        emit RevokedPermissionsInstant(msg.sender, from, permissionIds);
+    }
+
+    function _stageGrantPermission(address to, uint8 permissionId) private {
+        require(!_isStagedToCommit(), "Already staged");
+        uint256 diff = _permissionIdToMask(permissionId);
+        if (!_stagedAddresses.contains(to)) {
+            _stagedAddresses.add(to);
+            _stagedPermissionMasks[to] = _permissionMasks[to];
+        }
+        uint256 currentMask = _stagedPermissionMasks[to];
+        require(currentMask & diff == 0, ExceptionsLibrary.INVALID_TARGET);
+        _stagedPermissionMasks[to] = currentMask | diff;
+    }
+
+    function _rollbackStagedPermissions() internal {
+        require(_isStagedToCommit(), ExceptionsLibrary.INVARIANT);
+        _clearStagedPermissions();
+        delete _stagedToCommitAt;
+        emit RolledBackStagedPermissions(msg.sender);
+    }
+
+    function _stageGrantPermissions(
+        address to,
+        uint8[] calldata permissionIds,
+        uint256 delay
+    ) internal {
+        require(!_isStagedToCommit(), ExceptionsLibrary.INVARIANT);
+        for (uint256 i; i != permissionIds.length; ++i) {
+            _stageGrantPermission(to, permissionIds[i]);
+        }
+        _stagedToCommitAt = block.timestamp + delay;
+        emit StagedGrantPermissions(msg.sender, to, permissionIds, delay);
+    }
+
+    function _commitStagedPermissions() internal {
+        require(_isStagedToCommit(), ExceptionsLibrary.INVARIANT);
+        require(block.timestamp >= _stagedToCommitAt, ExceptionsLibrary.TIMESTAMP);
+        uint256 length = _stagedAddresses.length();
+        for (uint256 i; i != length; ++i) {
+            address delayedAddress = _stagedAddresses.at(i);
+            uint256 delayedPermissionMask = _stagedPermissionMasks[delayedAddress];
+            if (delayedPermissionMask == 0) {
+                delete _permissionMasks[delayedAddress];
+                _addresses.remove(delayedAddress);
+            } else {
+                _permissionMasks[delayedAddress] = delayedPermissionMask;
+                _addresses.add(delayedAddress);
+            }
```

Comment:
I think this could be refactored to
```
_permissionMasks[delayedAddress] = delayedPermissionMask;
      if (delayedPermissionMask == 0) {
          _addresses.remove(delayedAddress);
      } else {
          _addresses.add(delayedAddress);
      }
```